src/lib/storage/value_vector.hpp

```diff
@@ -0,0 +1,210 @@
+      : _string_length(other._string_length), _chars(other._chars, alloc) {}
+
+  // Add a string to the end of the vector
+  void push_back(const std::string& string);
+
+  // Return the value at a certain position.
+  const std::string operator[](const size_t n) const;
+
+  FixedString at(const ChunkOffset chunk_offset);
+
+  // TODO(team_btm): move??
+  class iterator : public boost::iterator_facade<iterator, FixedString, std::random_access_iterator_tag, FixedString> {
+   public:
+    iterator(size_t string_length, const pmr_vector<char>& vector, size_t pos = 0)
+        : _string_length(string_length), _chars(vector), _pos(pos) {}
+    iterator& operator=(const iterator& other) {
+      DebugAssert(_string_length == other._string_length && &_chars == &other._chars,
+                  "can't convert pointers from different vectors");
+      _pos = other._pos;
+      return *this;
+    }
+
+   private:
+    using facade = boost::iterator_facade<iterator, FixedString, std::random_access_iterator_tag, FixedString>;
+    friend class boost::iterator_core_access;
+    bool equal(iterator const& other) const { return this->_pos == other._pos; }
+    typename facade::difference_type distance_to(iterator const& other) const {
+      return (int64_t(other._pos) - int64_t(this->_pos)) / int64_t(_string_length);
+    }
+    void advance(typename facade::difference_type n) { _pos += n * _string_length; }
+    void increment() { _pos += _string_length; }
+    void decrement() { _pos -= _string_length; }
+    FixedString dereference() const { return FixedString(const_cast<char*>(&_chars[_pos]), _string_length); }
+
+    const size_t _string_length;
+    const pmr_vector<char>& _chars;
+    size_t _pos;
+  };
+
+  // Make the ValueVector of FixedStrings iterable in different ways
+  iterator begin() noexcept;
+  iterator end() noexcept;
+  iterator begin() const noexcept;
+  iterator end() const noexcept;
+
+  iterator cbegin() const noexcept;
+  iterator cend() const noexcept;
+
+  typedef boost::reverse_iterator<iterator> reverse_iterator;
+  reverse_iterator rbegin() noexcept;
+  reverse_iterator rend() noexcept;
+
+  // Return the number of entries in the column.
+  size_t size() const;
+
+  // Return the amount of allocated memory
+  size_t capacity() const;
+
+  // Request the vector capacity to be at least enough to contain n elements
+  void reserve(const size_t n);
+
+  // Remove elements from the vector
+  void erase(const iterator start, const iterator end);
+
+  // Reduce capacity to fit its size
+  void shrink_to_fit();
+
+  // Return a copy of the allocator object associated with the vector of values
+  PolymorphicAllocator<FixedString> get_allocator();
+
+  // Return a pointer to the first element of the char vector
+  const char* data() const;
+
+  // Return the calculated size of ValueVector in main memory
+  size_t data_size() const;
+
+ protected:
+  const size_t _string_length;
+  pmr_vector<char> _chars;
```

Comment:
Is there a mechanism to ensure that existing FixedStrings referring to a portion of _chars is not invalidated if a resize() operation (i.e. due to a push_back()) causes the reallocation of the pmr_vectors internal data array at a different address? If not, the implementation of FixedString seems rather fragile and storing a shared_ptr to _chars together with the appropriate offset might be more robust.